src/main/java/jenkins/branch/CustomOrganizationFolderDescriptor.java

```diff
@@ -150,7 +150,9 @@ private static void doAddSpecificDescriptors() {
             }
         }
         LOGGER.log(Level.FINE, "clearing {0}", old);
-        all().removeAll(old);
+        for (CustomOrganizationFolderDescriptor d : old) {
+            all().remove(d);
```

Comment:
:bug: can you show that this will not cause an infinite loop of death as you are now calling https://github.com/jenkinsci/jenkins/blob/6718a71f4be8cda7c575ae40ea0befec5c2ff1cf/core/src/main/java/hudson/ExtensionList.java#L197-L205